### PR TITLE
Android: Expose fast disc speed option

### PR DESCRIPTION
**Pull request by t895**
This is a bit of a weird PR since this setting has a history of being enabled without considering the consequences. This issue will likely be amplified by being made accessible to Android users, but it still can provide benefits to some games. So I've changed the description a bit to make the warning as clear as possible. 

Specifically @JosJuice wanted this setting only as a dedicated PR.

### CODE REVIEW

File: Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/model/BooleanSetting.java [hunk 1]
```diff
@@ -270,6 +271,7 @@ public enum BooleanSetting implements AbstractBooleanSetting
           MAIN_MMU,
           MAIN_PAUSE_ON_PANIC,
           MAIN_ACCURATE_CPU_CACHE,
+          MAIN_FAST_DISC_SPEED,
```
Comment (JosJuice) [inline Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/model/BooleanSetting.java:274]: Changing this while emulation is running should work fine.

**Review by JosJuice — COMMENTED** [review]
I don't like having this available as a global setting. I won't merge it, but if others think it should be merged, I won't block it.

**Comment by t895** [comment]
> I won't merge it, but if others think it should be merged, I won't block it.
How would you prefer this be presented? Like, only in the per-game settings? Or would you need a better warning? (Maybe I could have a dialog pop up for settings like these)

**Comment by JosJuice** [comment]
I'd rather just let someone else decide, because I have pretty heavy preconceived notions on this setting. (The very first piece of Dolphin code I touched, back in 2014, was for disc drive timing! I formed my opinion on this setting back then and haven't really reconsidered much since.)

But at least regarding your warning suggestion, I don't think adding more warnings on top of what you currently have would make much of a difference.

**Comment by t895** [comment]
> I'd rather just let someone else decide, because I have pretty heavy preconceived notions on this setting.

Uhm, what is your opinion then?

> But at least regarding your warning suggestion, I don't think adding more warnings on top of what you currently have would make much of a difference.

That's probably reasonable

**Comment by JosJuice** [comment]
> Uhm, what is your opinion then?

That it shouldn't be in the global settings. To be clear, I formed this opinion back when Android basically wasn't a thing, and I find it difficult to objectively think through what I actually think would be the most reasonable in the context of Android. That's why I would like someone else to make the decision.

**Comment by mbc07** [comment]
To be fair, [I requested a global toggle for this option on desktop years ago](https://bugs.dolphin-emu.org/issues/10127) and the overall consensus was that it shouldn't be global. As I already expressed in the issue tracker, I don't agree with this reasoning and would rather have it as a global toggle, however, if this is coming to Android, it should probably be kept per-game instead of being a global toggle, to keep the UI in sync with the desktop version...

**Comment by t895** [comment]
There isn't really a concept of per-game only settings on Android (unless I missed it somehow). What's the reasoning for having that in Qt?

**Comment by JosJuice** [comment]
The whole design of the per-game settings in DolphinQt is a legacy holdover.

**Comment by t895** [comment]
Oh, then I'm perfectly fine with this as is then.

**Comment by mbc07** [comment]
> There isn't really a concept of per-game only settings on Android (unless I missed it somehow)

Are you sure? Holding a game in the game list and selecting "Edit game config" works just like the per-game settings on Qt AFAICT...

**Comment by t895** [comment]
> Are you sure? Holding a game in the game list and selecting "Edit game config" works just like the per-game settings on Qt AFAICT...

You're right but fast disc speed is only accessible in the per game settings for Qt

**Comment by JosJuice** [comment]
> Are you sure? Holding a game in the game list and selecting "Edit Game Settings" works just like the per-game settings on Qt AFAICT...

I think you missed the key word here: "only". On Android, all settings can be set both on a global basis and per-game basis, but there aren't any settings that can only be set on a per-game basis.

**Comment by mbc07** [comment]
Fair enough. If there aren't any plans to have per-game only settings on Android, having fast disc speed as a global option LGTM.

(I would really like a global toggle for fast disc speed on Qt as well, but that's a different can of worms and out of scope for this PR)

**Pull request closed without merging**